### ISSUE TITLE
Uncaught TypeError: Cannot read property 'isEmpty' of undefined

### ISSUE THREAD

**Issue by lierdakil** [atom/atom#7269]
`TextEditor.setCursorBufferPosition` with an argument like this: `{row: '0', column:'0'}` produces the following error.

I gather that point fields *have* to be `int`s, but this error is absolutely non-descriptive. I believe  `validatePoint` should be inserted somewhere (personally, I'd put it somewhere in constructor and getters/setters for Point object, but that's probably just me).

**Atom Version**: 0.209.0-d965eb7
**System**: "Gentoo Base
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'isEmpty' of undefined

```
At /home/livid/bin/atom-dir/share/atom/resources/app.asar/src/selection.js:161

TypeError: Cannot read property 'isEmpty' of undefined
  at Selection.module.exports.Selection.isEmpty (/home/livid/bin/atom-dir/share/atom/resources/app.asar/src/selection.js:161:35)
  at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/home/livid/bin/atom-dir/share/atom/resources/app.asar/src/text-editor-component.js:187:87)
  at /home/livid/bin/atom-dir/share/atom/resources/app.asar/src/text-editor-component.js:298:28
  at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/home/livid/bin/atom-dir/share/atom/resources/app.asar/src/view-registry.js:168:9)
  at /home/livid/bin/atom-dir/share/atom/resources/app.asar/src/view-registry.js:3:61

```

### Commands

```
     -4:39.2.0 git-plus:add (atom-text-editor.editor.is-focused)
     -4:39.2.0 core:select-all (atom-text-editor.editor.is-focused)
     -4:37 git-plus:commit (atom-text-editor.editor.is-focused)
     -4:33.3.0 core:save (atom-text-editor.editor.is-focused)
     -4:29.8.0 terminal-panel:toggle (atom-text-editor.editor.is-focused)
     -4:28.2.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -4:28.2.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -4:26.3.0 command-palette:toggle (atom-text-editor.editor.is-focused)
     -4:23.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -4:23.9.0 git-plus:push (atom-text-editor.editor)
     -4:20.2.0 terminal-panel:toggle (atom-text-editor.editor.is-focused)
  2x -4:15.9.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -4:13.9.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -4:06.8.0 terminal-panel:toggle (atom-text-editor.editor.mini.is-focused)
  4x -1:34.8.0 core:close (atom-text-editor.editor.is-focused)
     -1:31.9.0 window:toggle-dev-tools (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "atom-lint",
      "haskell-hindent",
      "markdown-preview-plus",
      "markdown-preview",
      "autocomplete",
      "terminal-status",
      "linter-hlint"
    ]
  },
  "editor": {
    "showIndentGuide": true,
    "autoIndentOnPaste": false,
    "invisibles": {},
    "fontFamily": "DejaVu Sans Mono",
    "fontSize": 23,
    "softWrapAtPreferredLineLength": true,
    "scrollPastEnd": true
  }
}
```

### Installed Packages

```coffee
# User
autocomplete-haskell, v0.4.4
autocomplete-paths, v1.0.2
autocomplete-plus-python-jedi, v0.3.6
build, v0.37.0
git-plus, v5.1.2
haskell-ghc-mod, v0.7.5
haskell-pointfree, v0.1.2
ide-haskell, v0.5.12
language-dot, v1.6.0
language-haskell, v1.1.1
language-latex, v0.6.1
linter, v0.12.7
linter-coffeelint, v0.3.2
linter-plus, v0.4.0
linter-pylint, v0.2.2
markdown-preview-pandoc, v0.0.14
markdown-table-formatter, v2.5.1
markdown-writer, v1.4.0
terminal-panel, v1.14.1
zotero-bibtex-autocomplete, v0.0.0

# Dev
autocomplete-plus, v2.3.0
```

**Comment by lierdakil** [comment]
Furthermore, this breaks editor on which it was called.

**Comment by mnquintana** [comment]
I agree that the error could be more descriptive, but the API docs do explicitly say that [row and column need to be `Number`s.](https://atom.io/docs/api/v0.209.0/Point#instance-row)

**Comment by lierdakil** [comment]
As I said before, calling this on an editor breaks it -- it stops responding and starts spewing errors on any interaction -- so this goes a little bit deeper than just non-descriptive error message.

Furthermore, JS can sometimes implicitly convert to `String` somewhat unexpectedly, and it's not always easy to trace. While that could be probably solved (more like monkey-patched) by calling `parseInt` on any of `Point` arguments, that seems a little bit excessive.

**Comment by maxbrunsfeld** [comment]
Thanks for the report. I think this has already been fixed, and the fix will ship in v0.210. See #7077, atom/text-buffer#75, atom/text-buffer#76.

**Comment by lierdakil** [comment]
I'll wait for v0.210, but I'm not sure if mentioned PRs fix this particular problem, judging by stacktrace. I'll reopen if able to reproduce in v0.210.
